packages/extension-driver-ksqldb/src/lib/restfulClient.ts

```diff
@@ -0,0 +1,214 @@
+};
+
+const RESTFUL_API = {
+  INFO: '/info',
+  KSQL: '/ksql',
+  QUERY: '/query',
+};
+
+interface KsqlInfoResponse {
+  KsqlServerInfo: {
+    version: string;
+    kafkaClusterId: string;
+    ksqlServiceId: string;
+    serverStatus: string;
+  };
+}
+
+export type Header = {
+  header: {
+    queryId: string;
+    schema: string;
+  };
+};
+
+export type Row = {
+  row: {
+    columns: any[];
+  };
+};
+
+export type FinalMessage = {
+  finalMessage: string;
+};
+
+/**
+ * The example query response like below:
+ * [
+ *  {
+ *    "header": {
+ *      "queryId": "transient_RIDERLOCATIONS_356492705638097482",
+ *      "schema": "`PROFILEID` STRING, `LATITUDE` DOUBLE, `LONGITUDE` DOUBLE"
+ *    }
+ *  },
+ *  { "row": {"columns": ["c2309eec",37.7877,-122.4205]} },
+ *  { ...more rows },
+ *
+ *  { "finalMessage": "Query Completed"}
+ * ]
+ */
+export type QueryResponse = Header | Row | FinalMessage;
+
+export interface RestfulClientOptions {
+  host: string;
+}
+
+export class RestfulClient {
+  private options: RestfulClientOptions;
+  public client?: http2.ClientHttp2Session;
+  public connected = false;
+  private startSession: () => null | Promise<void> = () => null;
+
+  constructor(options: RestfulClientOptions) {
+    this.options = options;
+    this.connect();
+  }
+
+  public connect() {
+    this.startSession = () =>
+      new Promise((resolve, reject) => {
+        this.client = http2.connect(this.options.host);
+
+        this.client.on('connect', () => {
+          this.connected = true;
+          resolve();
+        });
+
+        this.client.on('error', (error: any) => {
+          reject(error);
+        });
+      });
+  }
+
+  public close(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      if (this.client) {
+        !this.client.destroyed && this.client.destroy();
+        this.connected = false;
+        resolve();
+      } else {
+        reject(new Error('Client is not initialized'));
+      }
+    });
+  }
+
+  public async checkConnection(): Promise<
+    KsqlInfoResponse['KsqlServerInfo']['serverStatus']
+  > {
+    try {
+      const res = await this.request<KsqlInfoResponse>(RESTFUL_API.INFO, 'GET');
+      return res.KsqlServerInfo['serverStatus'];
+    } catch (e) {
+      throw new Error('KsqlDb server is not ready');
+    }
+  }
+
+  public async checkConnectionRunning(): Promise<boolean> {
+    try {
+      const status = await this.checkConnection();
+      const isRunning = status === 'RUNNING';
+      return isRunning;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  public async query({
+    query,
+    query_params = {},
+  }: {
+    query: string;
+    query_params?: Record<string, any>;
+  }) {
+    // bind query parameters
+    const ksql = this.bindParams(query, query_params);
+
+    const buffer = Buffer.from(JSON.stringify({ ksql }));
+    const res = await this.request(RESTFUL_API.QUERY, 'POST', buffer);
+    return res;
+  }
+
+  public async exec({
```

Comment:
After discussion, the `exec` could be used to create/drop a table, and insert, update, and delete actions, but select and print commands are not allowed, maybe we could add comment to describe it

Reply:
I have added the comment in [03a2ffa](https://github.com/Canner/vulcan-sql/pull/255/commits/03a2ffaea6fa104d8d643e118ba1cb32bc13fa97).